nengo/dists.py

```diff
@@ -503,12 +503,27 @@ class CosineSimilarity(SubvectorLength):
 
     This is also equivalent to the distribution of a single coefficient from a
     unit vector (a single dimension of ``UniformHypersphere(surface=True)``).
+    Furthermore, ``CosineSimilarity(d+2)`` is equivalent to the distribution of a
+    single coordinate from points uniformly samples from the d-dimensional unit
+    ball (a single dimension of
+    ``UniformHypersphere(surface=False).sample(n, d)``). These relationships
+    have been detailed in [Voelker2017]_.
 
     This can be used to calculate an intercept ``c = ppf(1 - p)`` such that
     ``dot(u, v) >= c`` with probability ``p``, for random unit vectors ``u``
     and ``v``. In other words, a neuron with intercept ``ppf(1 - p)`` will
     fire with probability ``p`` for a random unit length input.
 
+    .. [Voelker2017]
+       `Aaron R. Voelker, Jan Gosmann, and Terrence C. Stewart.
+       Efficiently sampling vectors and coordinates from the n-sphere and
+       n-ball.
+       Technical Report, Centre for Theoretical Neuroscience, Waterloo, ON,
+       2017
+       <https://www.researchgate.net/publication/312056739_Efficiently_sampling_vectors_and_coordinates_from_the_n-sphere_and_n-ball>`_,
```

Comment:
Yes, but I don't think that's fixable. because it is in docstring it is also awkward to add a `# noqa` comment. I actually vote to disable the pyflakes line length check, but enable the it in pylint. pylint supports to define a regex for lines that are allowed to exceed the maximum length which could be configured to allow long URLs. I suppose I'll do a separate PR for that. (related: #1278)